plugin/http/propagation/b3/b3.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package b3 contains a propagation.HTTPFormat implementation
+// for B3 propagation. See https://github.com/openzipkin/b3-propagation
+// for more details.
+package b3
+
+import (
+	"encoding/hex"
+	"net/http"
+
+	"go.opencensus.io/trace"
+	"go.opencensus.io/trace/propagation"
+)
+
+const (
+	traceIDHeader = "X─B3─TraceId"
+	spanIDHeader  = "X─B3─SpanId"
+	sampledHeader = "X─B3─Sampled"
+)
+
+// HTTPFormat implements propagation.HTTPFormat to propagate
+// traces in HTTP headers in B3 propagation format.
+// HTTPFormat skips the X-B3-ParentId and X-B3-Flags headers
+// because there are additional fields not represented in the
+// OpenCensus span context. Spans created from the incoming
+// header will be the direct children of the client-side span.
+// Similarly, reciever of the outgoing spans should use client-side
+// span created by OpenCensus as the parent.
+type HTTPFormat struct{}
+
+var _ propagation.HTTPFormat = (*HTTPFormat)(nil)
+
+// FromRequest extracts a B3 span context from incoming requests.
+func (f *HTTPFormat) FromRequest(req *http.Request) (sc trace.SpanContext, ok bool) {
```

Comment:
Why use a pointer receiver? I'm just curious to know if there's a reason to prefer them in general even for trivial cases like this?

Reply:
I used the pointer receiver because I cannot foresee whether there will be a HTTPFormat with a field we want to mutate in its methods. I want us to be consistent, not requiring a concrete type at one place and the pointer at another place.